sys/crypto/chacha20poly1305.c

```diff
@@ -51,63 +51,47 @@ static void _r(uint32_t *a, uint32_t *b, uint32_t *d, unsigned c)
     *d = (tmp << c) | (tmp >> (32 - c));
 }
 
-static void _add_initial(chacha20poly1305_ctx_t *ctx, const uint8_t *key,
+static void _create_initial(chacha20_ctx_t *ctx, const uint8_t *key,
                          const uint8_t *nonce, uint32_t blk)
 {
     for (unsigned i = 0; i < 4; i++) {
-        ctx->state[i] += constant[i];
+        ctx->constant[i] = constant[i];
     }
     for (unsigned i = 0; i < 8; i++) {
-        ctx->state[i+4] += unaligned_get_u32(key + 4*i);
+        ctx->key[i] = unaligned_get_u32(key + 4*i);
     }
-    ctx->state[12] += unaligned_get_u32((uint8_t*)&blk);
-    ctx->state[13] += unaligned_get_u32(nonce);
-    ctx->state[14] += unaligned_get_u32(nonce+4);
-    ctx->state[15] += unaligned_get_u32(nonce+8);
+    ctx->counter[0] = unaligned_get_u32((uint8_t*)&blk);
+    ctx->nonce[0] = unaligned_get_u32(nonce);
+    ctx->nonce[1] = unaligned_get_u32(nonce+4);
+    ctx->nonce[2] = unaligned_get_u32(nonce+8);
 }
 
-static void _keystream(chacha20poly1305_ctx_t *ctx, const uint8_t *key,
-                       const uint8_t *nonce, uint32_t blk)
+static void _add_initial(chacha20_ctx_t *ctx, uint32_t *key_stream)
+{
+    for (unsigned i = 0; i < 16; i++) {
+        key_stream[i] += ((uint32_t *)ctx)[i];
+    }
+}
+
+static void _keystream(chacha20_ctx_t *ctx, uint32_t *key_stream)
```

Comment:
While you are at it, maybe you could add a short documentation to these internal functions. Would at least ease the review :)